third_party/terraform/resources/resource_bigquery_table.go.erb

```diff
@@ -838,6 +907,41 @@ func expandTimePartitioning(configured interface{}) *bigquery.TimePartitioning {
 	return tp
 }
 
+<% unless version == 'ga' -%>
+func expandRangePartitioning(configured interface{}) (*bigquery.RangePartitioning, error) {
+	if configured == nil {
+		return nil, nil
+	}
+
+	rpJson := configured.([]interface{})
+	if len(rpJson) == 0 || rpJson[0] == nil {
+		return nil, errors.New("Error casting range partitioning interface to expected structure")
+	}
+
+	rpRaw := rpJson[0].([]map[string]interface{})
```

Comment:
I'm getting some type errors here, so this will need some tweaking. configured is a single-item list of RangePartition maps, which contains a single-item list of "range" maps. The real value we want is:

rangePartList := configured.([]interface{})
rangePartObj := rangePartList[0].(map[string]interface{})
rangeList := rangePart["range"].([]interface{})
rangeObj := rangeList[0].map[string]interface{}

so pretty much that with index/nil checking is what we want